messaging/src/main/java/org/axonframework/eventhandling/annotation/AnnotatedEventHandlingComponent.java

```diff
@@ -0,0 +1,150 @@
+        this(annotatedEventHandler,
+             parameterResolverFactory,
+             ClasspathHandlerDefinition.forClass(annotatedEventHandler.getClass()));
+    }
+
+    /**
+     * Wraps the given {@code annotatedEventHandler}, allowing it to be subscribed to a
+     * {@link org.axonframework.eventhandling.EventSink} as a {@link EventHandlingComponent}.
+     *
+     * @param annotatedEventHandler    The object containing the
+     *                                 {@link org.axonframework.eventhandling.annotation.EventHandler} annotated
+     *                                 methods.
+     * @param parameterResolverFactory The strategy for resolving handler method parameter values.
+     * @param handlerDefinition        The handler definition used to create concrete handlers.
+     */
+    @SuppressWarnings("unchecked")
+    public AnnotatedEventHandlingComponent(@Nonnull T annotatedEventHandler,
+                                           @Nonnull ParameterResolverFactory parameterResolverFactory,
+                                           @Nonnull HandlerDefinition handlerDefinition) {
+        this.target = requireNonNull(annotatedEventHandler, "The Annotated Event Handler may not be null");
+        this.model = AnnotatedHandlerInspector.inspectType((Class<T>) annotatedEventHandler.getClass(),
+                                                           parameterResolverFactory,
+                                                           handlerDefinition);
+    }
+
+    /**
+     * Wraps the given {@code annotatedEventHandler}, allowing it to be subscribed to a
+     * {@link org.axonframework.eventhandling.EventSink} as a {@link EventHandlingComponent}.
+     *
+     * @param annotatedEventHandler The object containing the
+     *                              {@link org.axonframework.eventhandling.annotation.EventHandler} annotated methods.
+     * @param model                 The inspector to use to find the annotated handlers on the annotatedEventHandler.
+     */
+    public AnnotatedEventHandlingComponent(@Nonnull T annotatedEventHandler,
+                                           @Nonnull AnnotatedHandlerInspector<T> model) {
+        this.target = requireNonNull(annotatedEventHandler, "The Annotated Event Handler may not be null");
+        this.model = requireNonNull(model, "The Annotated Handler Inspector may not be null");
+    }
+
+    @Override
+    public EventHandlerRegistry subscribe(@Nonnull QualifiedName name, @Nonnull EventHandler eventHandler) {
+        throw new UnsupportedOperationException(
+                "This Event Handling Component does not support direct event handler registration."
+        );
+    }
+
+    @Nonnull
+    @Override
+    public MessageStream.Empty<Message<Void>> handle(@Nonnull EventMessage<?> event,
+                                                     @Nonnull ProcessingContext context) {
+        requireNonNull(event, "Event Message may not be null");
```

Comment:
I am wondering whether it is this method's task to do a try-catch around the entire block, and move the failure as a `MessageStream#failed` or not.

Granted, I can imagine we would see the return of the `ErrorHandler` and `ListenerInvocationErrorHandler` that currently reside in the `EventProcessor` hierarchy to take care of exceptions as the user sees fit.

Just food for thought here, not something we need to resolve right now.